### PR TITLE
[orf] Update OE1 extractor, merge with FM4.

### PR DESCRIPTION
**Pull request by phaer**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Since oe1.orf.at has been updated, both ORF radios supported by youtube_dl
use the same API. This commit honors this fact by merging both extractors
into one, improving code re-use and making sure that oe1.orf.at will continue to work
whenever the legacy API gets deprecated.

flake8 has no complaints and the tests are passing.

### CODE REVIEW

File: youtube_dl/extractor/orf.py [hunk 1]
```diff
@@ -144,77 +142,60 @@ def quality_to_int(s):
...
-            'id': show_id,
-            'title': data['item']['title'],
-            'url': data['item']['url_stream'],
-            'ext': 'mp3',
-            'description': data['item'].get('info'),
-            'timestamp': unix_timestamp
-        }
-
-
-class ORFFM4IE(InfoExtractor):
```
Comment (dstftw) [inline youtube_dl/extractor/orf.py]: Original extractor classes (`ie_key`s) should be preserved. Otherwise you break existing download archives.

Comment (midas02) [inline youtube_dl/extractor/orf.py]: @dstftw I was looking at fixing this, but:

1. In this commit it seems to me you have removed a class yourself. What's the difference with this situation? https://github.com/rg3/youtube-dl/commit/6d1ded75021ca76f14c4fe8d3aa698704bae1dd9
2. What do you mean by 'you break existing download archives'?

Comment (dstftw) [inline youtube_dl/extractor/orf.py]: 1. There is no other way in this case. pluzz.francetv.fr does not exist anymore.
2. `ie_key`s in `--download-archive` file.

Comment (phaer) [inline youtube_dl/extractor/orf.py]: Sorry, it took me a while to come back to this pull request. My latest commit just re-introduces the old classes which are now simply extending their common base class. This should preserve `ie_key`s. What do you think, @dstftw ?

**Pull request closed without merging**